How to Fork and Create Pull Requests

SaintPeter opened the pull request:
Two articles on how to create and maintain a local fork and how to submit a PR to FCC.

Request Review by @ltegman and @bugron for completeness.

Rafase282 commented:
Looks good to me. I personally use the Github official windows app most of the time. But having it use the command line is much better.

bugron commented on How-To-Create-A-Pull-Request-for-Free-Code-Camp.md:
becuase -> because

bugron commented on How-To-Create-A-Pull-Request-for-Free-Code-Camp.md:
reccomended - > recommended

bugron commented on How-To-Create-A-Pull-Request-for-Free-Code-Camp.md:
reccomended - > recommended

bugron commented on How-To-Create-A-Pull-Request-for-Free-Code-Camp.md:
maintenence -> maintenance

bugron commented on How-To-Create-A-Pull-Request-for-Free-Code-Camp.md:
againt -> against

bugron commented on How-To-Create-A-Pull-Request-for-Free-Code-Camp.md:
sucinctly -> succinctly

bugron commented on How-To-Create-A-Pull-Request-for-Free-Code-Camp.md:
accpted -> accepted

bugron commented on How-To-Create-A-Pull-Request-for-Free-Code-Camp.md:
dispair -> despair

bugron commented on How-To-Fork-And-Maintain-a-Local-Instance-of-Free-Code-Camp.md:
availible -> available

bugron commented on How-To-Fork-And-Maintain-a-Local-Instance-of-Free-Code-Camp.md:
enviornment -> environment

bugron commented on How-To-Fork-And-Maintain-a-Local-Instance-of-Free-Code-Camp.md:
puposes -> purposes

bugron commented on How-To-Create-A-Pull-Request-for-Free-Code-Camp.md:
Maybe add a short note about `git add .` here?

bugron commented:
Awesome articles @SaintPeter!

ltegman commented on How-To-Fork-And-Maintain-a-Local-Instance-of-Free-Code-Camp.md:
I wouldn't say that all commands can be executed in a graphical Git environment, only most of them.

ltegman commented on How-To-Fork-And-Maintain-a-Local-Instance-of-Free-Code-Camp.md:
If it's not too much trouble it might be nice to have little screen snaps for these steps that ask a user to perform an action in the Github web UI.

ltegman commented on How-To-Create-A-Pull-Request-for-Free-Code-Camp.md:
Maybe it's worth it if it gets people to always be using branches, but I'm somewhat uncomfortable with making this sound so dire since recovering a staging branch is as simple as `git reset --hard HEAD~numOfBadCommits` and then force pushing. I feel like all too often people get stuck in git and think the only option is to delete and restart when it rarely is the only option.

ltegman commented:
Besides the couple inline notes from myself and @bugron I think this looks great :smile:. Will be good to have a solid starting people for people.

My only other thoughts are that perhaps (even though it's harder) we should put the recommended way to contribute (local copy) first. I think putting the easy but discouraged way first is a good way to end up with a lot of people going that route.

I don't think we should hold up these articles for it, but I also want to suggest that we create an article on resolving rebase conflicts and link to that from one (or both) of these articles. I'm happy to do that within the next week if no one else does.

SaintPeter commented on How-To-Create-A-Pull-Request-for-Free-Code-Camp.md:
With regards to editing on a branch and fixing a broken branch . . .
I agree in principle that it is possible to recover, but it is better for a new contributor to have the holy fear of god put into them that they always edit on a branch than to believe it is ever acceptable to edit on their master/staging.  By the time they learn it is not true, they will be skilled enough to fix problems themselves.

bugron commented on How-To-Create-A-Pull-Request-for-Free-Code-Camp.md:
"unchanged" or "unstaged"?

SaintPeter commented:
Thanks @bugron for your excellent copy editing.  I kinda wrote it in a hurry, but in reality I'm a lousy speller no matter how long I take.  Without a spellcheck I am lost.  You'll notice that at least I'm consistent in my misspellings :wink:.

@ltegman I reversed the order of the Local Copy vs. Web Interface and added a "Recommended" tag.  I agree, we should encourage users to do that first.

RE: Rebase Conflicts
OMG, yes, please.  I need that one as much as anyone.

NOTE: 
I also added my new "HelpContributors" channel as a link at the top of each document.  My hope is that we can offer direct support to folks considering doing a PR.

Once these are live, I'll also add links to the README.md and CONTRIBUTING.md to these documents.

I wonder if we shouldn't also have a "So you want to contribute" type doc that acts as an overview?
Maybe I'll just link everything from the README.md, in the hopes that we'll get more people there first.

---
Anyway, I think I got all the spelling errors.  Give it a quick look see and we can get this live.

SaintPeter commented:
LOL, @bugron, yes, thank you. :smile: Corrected.

bugron commented:
Excellent work, @SaintPeter, love these articles, merging to have them live ASAP.
